app/actions/remote/entry/notification.ts

```diff
@@ -120,8 +120,12 @@ export const pushNotificationEntry = async (serverUrl: string, notification: Not
     let removeTeams;
     if (removeTeamIds?.length) {
         // Immediately delete myTeams so that the UI renders only teams the user is a member of.
-        removeTeams = await queryTeamsById(operator.database, removeTeamIds);
-        await deleteMyTeams(operator, removeTeams!);
+        const removeMyTeams = await queryMyTeamsById(database, removeTeamIds);
+        if (removeMyTeams?.length) {
+            await deleteMyTeams(operator, removeMyTeams);
+            const ids = removeMyTeams.map((m) => m.id);
+            removeTeams = await queryTeamsById(database, ids);
+        }
```

Comment:
This code is repeated 3x times.  0/5 if a small helper function would be useful.